src/nvim/terminal.c

```diff
@@ -1217,12 +1218,14 @@ static bool is_focused(Terminal *term)
 
 #define GET_CONFIG_VALUE(k, o) \
   do { \
-    Error err; \
+    Error err = ERROR_INIT; \
     /* Only called from terminal_open where curbuf->terminal is the */ \
     /* context  */ \
     o = dict_get_value(curbuf->b_vars, cstr_as_string(k), &err); \
+    api_free_error(err); \
     if (o.type == kObjectTypeNil) { \
       o = dict_get_value(&globvardict, cstr_as_string(k), &err); \
+      api_free_error(err); \
```

Comment:
This call can result in a double free:
If the key is not found in `curbuf->b_vars` then `err` is set and freed by the previous call to `api_free_error()`.
But `err->msg` still points to the freed memory.
If the key is found in `globalvardict` then no error is set but an invalid memory address is freed.